packages/flutter_tools/gradle/src/main/groovy/flutter.groovy

```diff
@@ -390,19 +390,63 @@ class FlutterPlugin implements Plugin<Project> {
+    //  which includes all the plugins from the `.flutter-plugins` file, even if not made for Android. 
+    //  The settings.gradle then:
+    //      1) tries to add the android plugin implementation, which does not exist at all, but is also not included
+    //         successfully (which does not throw an error and therefore isn't a problem), or
+    //      2) includes the plugin successfully as a valid android plugin directory exists, even if the surrounding
+    //         flutter package does not support the android platform (see e.g. apple_maps_flutter: 1.0.1). 
+    //         So as it's included successfully it expects to be added as API. This is only possible by taking all 
+    //         plugins into account, which only appear on the `dependencyGraph` and under the `plugins`.
+    //  So in summary the plugins are currently selected from the `dependencyGraph` and filtered then with the
+    //  [doesSupportAndroidPlatform] method instead of just using the `plugins.android` list.
+    private configureLegacyPluginProjects() {
+        File settingsGradle = new File(project.projectDir.parentFile, 'settings.gradle')
+        if(!settingsGradle.exists() || !settingsGradle.text.contains("'.flutter-plugins'")) {
+            return
+        }
+        project.logger.error("Your app uses an outdated version of `settings.gradle`. Please update.")
+        List deps = getPluginDependencies()
+        List plugins = getPluginList().collect { it.name }
+        deps.removeIf { plugins.contains(it.name) }
+        deps.each {
+            Project pluginProject = project.rootProject.findProject(":${it.name}")
+            if (pluginProject != null &&
+                    doesSupportAndroidPlatform(pluginProject.projectDir.parentFile.path)) {
+                configurePluginProject(it)
+            }
+        }
+    }
+
+    // TODO(gustl22): Can remove this check, once #54566 is resolved.
+    /**
+     * Returns `true` if the given path contains an `android/build.gradle` file.
+     */
+    private static Boolean doesSupportAndroidPlatform(String path) {
+        File editableAndroidProject = new File(path, 'android' + File.separator + 'build.gradle')
+        return editableAndroidProject.exists()
     }
 
     /** Adds the plugin project dependency to the app project. */
-    private void configurePluginProject(String pluginName, String _) {
-        Project pluginProject = project.rootProject.findProject(":$pluginName")
+    private void configurePluginProject(Object pluginObject) {
+        assert pluginObject.name instanceof String
+        Project pluginProject = project.rootProject.findProject(":${pluginObject.name}")
+
         if (pluginProject == null) {
-            project.logger.error("Plugin project :$pluginName not found. Please update settings.gradle.")
+            project.logger.error("Plugin project :${pluginObject.name} not found. Please update settings.gradle.")
```

Comment:
How should the developer update settings.gradle?

Reply:
related https://github.com/flutter/flutter/issues/135392

Reply:
I rechecked with the previous code and the error was thrown in these two cases:

- Using a more recent `settings.gradle`, but loading an unsupported plugin, e.g. Plugin provides a working `android` folder, but provides no support for android in pubspec file, means no entry for `flutter: -> plugin: -> platforms: -> android:`, which this PR is fixing, as no plugins are loaded any more, which are not supported
- Using an old `PluginsEach` settings gradle, which was altered by the developer to not include plugins, which is basically the same as using a more recent `settings.gradle` (above case).

Reply:
Moved via b7901cd7a9f9ca88364202aa6c89461eaecc1049